front/src/pages/mainPage/components/Search.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { TypeProduct } from '../../../types';
+import { getSearchedProductNames } from '../../../axios';
+import { useThrottle, useDebounce } from '../../../hooks';
+import { Link } from 'react-router-dom';
+import SearchBar from './Searchbar';
+
+
+const SearchComponent: React.FC = () => {
+    const [query, setQuery] = useState<string>('');
+    const [results, setResults] = useState<TypeProduct[]>([]);
+    const [loading, setLoading] = useState<boolean>(false);
+
+    const fetchData = async () => {
+        getSearchedProductNames(query, setResults, setLoading)
+    };
+
+    const debounceValue = useDebounce(query, 2000);
+    const throttledFetch = useThrottle(fetchData, 5000);
+
+    useEffect(() => {
+        console.log("Debounced:", query);
+        fetchData();
+    }, [debounceValue]);
+
+
+    return (
+        <div className="search_holder">
+            <SearchBar throttledFetch={throttledFetch} query={query} setQuery={setQuery} />
+            <div className='search_data'>
+                {loading ? (
+                    <p>Loading...</p>
+                ) : (
+                    <ul>
```

Comment:
Results of the Search should be displayed in the MainContent, absolutely the same way it looks when specific category is displayed.

Responsibility of SearchBar / ProductSearch should be limited to querying right products and bypassing this data to the MainContent (Similarly how selecting of category in the sidebar works - Sidebar menu doesn't know how to render individual product results)